app/controllers/file.rb

```diff
@@ -13,71 +13,83 @@ module Controllers
 		class File < Base
 			helpers Tint::Helpers::Rendering
 
-			namespace "/:site/files" do
-				get "/?*" do
-					file = site.file(params['splat'].join('/'))
+			def self.query(val)
+				condition { request.query_string == val }
+			end
 
-					if params[:download] && file.exist? && !file.directory?
-						authorize file, :show?
-						return send_file file.path, filename: file.name, type: file.mime, disposition: :attachment
+			namespace "/:site/files" do
+				get "/?*", query: "download" do
+					if resource.exist? && resource.file?
+						authorize resource, :show?
+						send_file resource.path, filename: resource.name, type: resource.mime, disposition: :attachment
+					else
+						slim :error, locals: { message: "Only files may be downloaded." }
 					end
+				end
 
-					if file.directory? || !file.exist?
-						authorize file.to_directory, :index?
-						render_directory file.to_directory
-					elsif file.text?
-						authorize file, :edit?
+				get "/?*", query: "source" do
+					authorize resource, :edit?
 
-						if params.has_key?('source')
-							source(file)
-						elsif file.yml? || !file.content?
-							slim :"layouts/files" do
-								slim :"files/yml", locals: {
-									data: file.frontmatter,
-									path: file.route
-								}
-							end
+					if resource.text?
+						html = slim :"layouts/files" do
+							slim :"files/source", locals: { path: resource.route }
+						end
+
+						stream_into_element("<textarea name=\"source\">", html, resource)
+					else
+						slim :error, locals: { message: "Only text files may be edited by source" }
+					end
+				end
+
+				get "/?*" do
+					if resource.directory? || !resource.exist?
```

Comment:
could we make all of these if conditions, just be    `condition: lambda { resource.directory? || !resource.exist? }` or similar?